teuthology/task/ceph_ansible.py

```diff
@@ -116,15 +173,145 @@ def generate_hosts_file(self):
         self.inventory = self._write_hosts_file(hosts_stringio.read().strip())
         self.generated_inventory = True
 
+    def begin(self):
+        super(CephAnsible, self).begin()
+        self.execute_playbook()
+
+    def _write_hosts_file(self, content):
+        """
+        Actually write the hosts file
+        """
+        hosts_file = NamedTemporaryFile(prefix="teuth_ansible_hosts_",
+                                        delete=False)
+        hosts_file.write(content)
+        hosts_file.flush()
+        return hosts_file.name
+
+    def wait_for_ceph_health(self):
+        with contextutil.safe_while(sleep=15, tries=6,
+                                    action='check health') as proceed:
+            (remote,) = self.ctx.cluster.only('mon.a').remotes
+            remote.run(args=['sudo', 'ceph', 'osd', 'tree'])
+            remote.run(args=['sudo', 'ceph', '-s'])
+            log.info("Waiting for Ceph health to reach HEALTH_OK \
+                        or HEALTH WARN")
+            while proceed():
+                out = StringIO()
+                remote.run(args=['sudo', 'ceph', 'health'], stdout=out)
+                out = out.getvalue().split(None, 1)[0]
+                log.info("cluster in state: %s", out)
+                if (out == 'HEALTH_OK' or out == 'HEALTH_WARN'):
+                    break
+
     def get_host_vars(self, remote):
         extra_vars = self.config.get('vars', dict())
         host_vars = dict()
         if not extra_vars.get('osd_auto_discovery', False):
-            host_vars['devices'] = get_scratch_devices(remote)
+            roles = self.ctx.cluster.remotes[remote]
+            dev_needed = len([role for role in roles
+                              if role.startswith('osd')])
+            host_vars['devices'] = get_scratch_devices(remote)[0:dev_needed]
         if 'monitor_interface' not in extra_vars:
             host_vars['monitor_interface'] = remote.interface
         if 'public_network' not in extra_vars:
             host_vars['public_network'] = remote.cidr
         return host_vars
 
+    def run_rh_playbook(self):
+        ceph_installer = self.ceph_installer
+        args = self.args
+        ceph_installer.run(args=['cp', '-R',
+                                 '/usr/share/ceph-ansible', '.'])
+        ceph_installer.put_file(self.inventory, 'ceph-ansible/inven.yml')
+        ceph_installer.put_file(self.playbook_file.name, 'ceph-ansible/site.yml')
+        ceph_installer.run(args=('cat', self.inventory))
```

Comment:
Did you mean to cat ``ceph-ansible/inven.yml`` here instead of ``self.inventory``?